### PR TITLE
PR #133 cleaned - Autosploit Automation

### PR DESCRIPTION
**Pull request by Selora**
#133 Rebased and cleaned. 

### PR Breakdown:
- Bash scripts wrappers around autosploit. Could be use to test against a whitelist of your organisation's IP addresses.
- Vagrantfile skeleton and provisioning file (used with lightsail provider). Environment dependant, use as a starting point.
- Removed fuzzers from default modules JSON file. The fuzzer modules would block the script's execution forever. They are moved into a new json file. They can be run using the interactive terminal or specifying the JSON explicitly with --exploit-file-to-use
- Bugfixes for PR #132 . Fail/Success counters fix.
- Improvements: Meterperter now starts in background, which allows an operator to run msfconsole into the corresponding workspace and interact with the new session while Autosploit is still finishing the remaining exploits in background. 

### Bash script usage:
**./dryrun_autosploit.sh whitelist.txt "search_query"**
Will search censys/shodan/etc and do a dry-run against discovered hosts that are in the whitelist.
This will generate a report containing the dry-run results for every hosts in the whitelist.
This does NOT exploit anything.

**./run_autosploit.sh**
Will run autosploit in exploit mode against previously discovered hosts in the whitelist.
This will generate a report with MSF results for every hosts in the whitelist.

### Important:

 -   You should do a dry-run before running "run_autosploit.sh".
 -   Make sure that the whitelist is sound with a given scope, and validate against the dry-run report.
 -   In other words, VALIDATE THE DRYRUN REPORT BEFORE LAUNCHING THE EXPLOIT RUN.

The code to compare each hosts against a whitelist **hasn't been thoroughly tested**. If you don't validate a dryrun report and run the "run_autosploit.sh" script, and you hit hosts that are outside your given scope, you have been warned. Run at your own risks.

### CODE REVIEW

**Comment by Ekultek** [comment]
Love it

**Comment by Ekultek** [comment]
Merging
